server/service/osquery_utils/queries.go

```diff
@@ -1297,23 +1303,65 @@ func directIngestChromeProfiles(ctx context.Context, logger log.Logger, host *fl
 func directIngestBattery(ctx context.Context, logger log.Logger, host *fleet.Host, ds fleet.Datastore, rows []map[string]string) error {
 	mapping := make([]*fleet.HostBattery, 0, len(rows))
 	for _, row := range rows {
-		cycleCount, err := strconv.ParseInt(EmptyToZero(row["cycle_count"]), 10, 64)
+		cycleCount, err := strconv.Atoi(EmptyToZero(row["cycle_count"]))
 		if err != nil {
 			return err
 		}
-		mapping = append(mapping, &fleet.HostBattery{
-			HostID:       host.ID,
-			SerialNumber: row["serial_number"],
-			CycleCount:   int(cycleCount),
-			// database type is VARCHAR(40) and since there isn't a
-			// canonical list of strings we can get for health, we
-			// truncate the value just in case.
-			Health: fmt.Sprintf("%.40s", row["health"]),
-		})
+
+		switch host.Platform {
+		case "darwin":
+			mapping = append(mapping, &fleet.HostBattery{
+				HostID:       host.ID,
+				SerialNumber: row["serial_number"],
+				CycleCount:   cycleCount,
+				// database type is VARCHAR(40) and since there isn't a
+				// canonical list of strings we can get for health, we
+				// truncate the value just in case.
+				Health: fmt.Sprintf("%.40s", row["health"]),
+			})
+		case "windows":
+			mapping = append(mapping, &fleet.HostBattery{
+				HostID:       host.ID,
+				SerialNumber: row["serial_number"],
+				CycleCount:   cycleCount,
+				Health:       generateWindowsBatteryHealth(row["designed_capacity"], row["max_capacity"]),
+			})
+		}
 	}
 	return ds.ReplaceHostBatteries(ctx, host.ID, mapping)
 }
 
+const (
+	batteryStatusUnknown     = "Unknown"
+	batteryStatusDegraded    = "Check Battery"
+	batteryStatusGood        = "Good"
```

Comment:
Are [osquery docs](https://osquery.io/schema/5.13.1/#battery) accurate?
```
One of the following: "Good" describes a well-performing battery, "Fair" describes a functional battery with limited capacity, or "Poor" describes a battery that's not capable of providing power
```

Reply:
good catch, i don't believe it's accurate:

on darwin, the value is passed through from Apple's [API](https://developer.apple.com/documentation/devicemanagement/statusdevicebatteryhealth), which doesn't match the docs

on windows, the `health` column is omitted

Reply:
Gotcha, data from dogfood:
```
SELECT DISTINCT health FROM fleet.host_batteries;
+---------------+
| health        |
+---------------+
| Good          |
| Check Battery |
| Fair          |
+---------------+
```

I think this looks good because it matches "Good" and "Check Battery" from macOS.

Maybe "Poor" in the osquery docs should be changed to "Check Battery"?

Reply:
`Fair` 👀